git: git reset with branch

chulkilee opened the pull request:
##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
test/integration/targets/git/tasks/main.yml

##### ANSIBLE VERSION
a6e8359

##### SUMMARY

Add test for ansible/ansible-modules-core#3386

`git reset <ref>` can be ambiguous and fail to switch to the correct branch. To avoid it, specify branch.

mscherer commented:
Could you rebase the PR ? I can't merge without it.

chulkilee commented:
Rebased.

mattclay commented:
@chulkilee Now that we have the modules and the tests in the same repository, would you be interested in rebasing this PR and adding the fix from https://github.com/ansible/ansible-modules-core/pull/3386 to it?

chulkilee committed:
git: git reset with branch

`git reset <ref>` can be ambiguous and fail to switch to the correct branch.
To avoid it, specify branch as well.

chulkilee commented:
@mattclay amended the commit.

mattclay commented:
@robinro Fix and tests are now in the same PR. Tests are passing. Does this look good to you?

robinro reviewed (approved):
Looks good.

Thanks again @chulkilee for the fix and waiting so long. @mattclay thanks for looking into it. Let's merge it.

robinro commented:
https://github.com/ansible/ansible-modules-core/pull/3386 should be closed when this gets merged.
